File: src/main/java/br/com/alisonviana/service/impl/StudentServiceImpl.java
package br.com.alisonviana.service.impl;

import br.com.alisonviana.entity.Student;
import br.com.alisonviana.repository.StudentRepository;
import br.com.alisonviana.service.StudentService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StudentServiceImpl implements StudentService {

    @Autowired
    private StudentRepository studentRepository;

    @Override
    public List<Student> getAllStudents() {
        return studentRepository.findAll();
    }

    @Override
    public Student getStudentById(Long id) {
        return studentRepository.findById(id).get();
    }

    @Override
    public Student saveStudent(Student student) {
        return studentRepository.save(student);
    }

    @Override
    public Student updateStudent(Student student) {
        return studentRepository.save(student);
    }

    @Override
    public void deleteStudentById(Long id) {
        studentRepository.deleteById(id);
    }

}

File: src/main/kotlin/br/com/alisonviana/service/impl/StudentServiceImpl.kt
package br.com.alisonviana.service.impl

import br.com.alisonviana.entity.Student
import br.com.alisonviana.repository.StudentRepository
import br.com.alisonviana.service.StudentService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class StudentServiceImpl(
    @Autowired
    var studentRepository: StudentRepository
): StudentService {
    override fun getAllStudents(): List<Student> =
        studentRepository.findAll()

    override fun getStudentById(id: Long): Student =
        studentRepository.findById(id).get()

    override fun saveStudent(student: Student): Student =
        studentRepository.save(student)

    override fun updateStudent(student: Student): Student =
        studentRepository.save(student)

    override fun deleteStudentById(id: Long) =
        studentRepository.deleteById(id)
}